docker/Dockerfile

```diff
@@ -40,10 +42,22 @@ FROM docker.io/python:${PYTHON_VERSION}-slim as requirements
 RUN \
    --mount=type=cache,target=/var/cache/apt,sharing=locked \
    --mount=type=cache,target=/var/lib/apt,sharing=locked \
- apt-get update && apt-get install -y git \
+ apt-get update && apt-get install -y \
+    git \
+    build-essential \
+    libffi-dev \
+    libjpeg-dev \
+    libpq-dev \
+    libssl-dev \
+    libwebp-dev \
+    libxml++2.6-dev \
+    libxslt1-dev \
+    openssl \
+    rustc \
+    zlib1g-dev \
     && rm -rf /var/lib/apt/lists/*
 
-# We install poetry in its own build stage to avoid its dependencies conflicting with
+# We install poetry in its own prefix to avoid its dependencies conflicting with
```

Comment:
I'm struggling to understand why this is a change we need to make now. If we absolutely do, then anoa's suggestion to use a venv does seem to feel a bit more plain to me